packages/cli/test/demo/names-in-transit.js

```diff
@@ -0,0 +1,41 @@
+import { withContext as withContextSendingMessages } from './sending-messages.js';
+
+export const section = async (execa, testLine) => {
+  await withContextSendingMessages(execa, async () => {
+    // In this example, we send alice our "doubler" but let it appear...
+    await testLine(
+      execa`endo send alice ${'Please enjoy this @counter:doubler.'}`,
+    );
+    await testLine(execa`endo inbox --as alice-agent`, {
+      stdout: /^1\. \"HOST\" sent \"Please enjoy this \@counter\.\"/,
+    });
+    await testLine(
+      execa`endo adopt --as alice-agent 1 counter --name redoubler`,
+    );
+    await testLine(execa`endo list --as alice-agent`, {
+      stdout: 'redoubler',
+    });
+    await testLine(execa`endo dismiss --as alice-agent 1`);
+  });
+};
+
+/**
+ * Wraps a function with the setup and teardown of the endo object state from names-in-transit.
+ *
+ * @param {*} execa
+ * @param {() => Promise<void} implementation
+ * @returns {Promise<void>}
+ */
+export const withContext = async (execa, implementation) => {
+  await withContextSendingMessages(execa, async () => {
+    try {
+      await execa`endo send alice ${'Please enjoy this @counter:doubler.'}`;
```

Comment:
I see that Agoric's ESLint rule `@jessie.js/safe-await-separator` is triggered here. Usually we get around this problem like so:
```suggestion
  await withContextSendingMessages(execa, async () => {
    await null;
    try {
      await execa`endo send alice ${'Please enjoy this @counter:doubler.'}`;
```

Async functions execute synchronously until they encounter their first `await`. `await null` is of course semantically meaningless, but it has the side effect of ceding control of the event loop, thus ending the "synchronous prelude" of this `async` function. The lint rule exists because it can otherwise be hard to reason about which parts of an `async` function are executed synchronously vs. asynchronously.

IMO, this is a false positive since no expressions or statements are evaluated before the first `await`. Nevertheless, if you want to silence these warnings, you can.

Reply:
interesting, thanks